chainer/functions/loss/hinge_max_margin.py

```diff
@@ -0,0 +1,181 @@
+            margin_{i}^{2} & {\\rm if~norm} = {\\rm Huber \\& margin_{i}<2}
+
+            \\end{array} \\right.
+
+        All 3 norms are continuous.
+        ``'L2'`` and ``'Huber'`` are differentiable, ``'L1'`` is not.
+
+        The output is a variable whose value depends on the value of
+        the option ``reduce``. If it is ``'along_second_axis'``, it holds the loss
+        values for each example. If it is ``'mean'``, it takes the mean of loss values.
+
+        See: `Huber loss - Wikipedia <https://en.wikipedia.org/wiki/Huber_loss>`
+        and Structured support vector machine - Wikipedia
+        <https://en.wikipedia.org/wiki/Structured_support_vector_machine>'
+
+    Args:
+        x (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray` of :class:`numpy.float`):
+            Input variable. The shape of ``x`` should be (:math:`N`, :math:`K`)
+            .
+        t (:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray` of signed integer):
+            The :math:`N`-dimensional label vector with values
+            :math:`t_n \\in \\{0, 1, 2, \\dots, K-1\\}`.
+            The shape of ``t`` should be (:math:`N`,).
+        norm (string): Specifies norm type. Either ``'L1'`` , ``'L2'`` , ``'Huber'`` are
+            acceptable.
+        reduce (str): Reduction option. Its value must be either
+            ``'mean'`` [default] or ``'along_second_axis'``.
+             Otherwise, :class:`ValueError` is raised.
+
+    Returns:
+        ~chainer.Variable:
+            A variable object holding the hinge max margin loss.
+            If ``reduce`` is ``'along_second_axis'``, the output variable holds array
+            whose shape is same :math:`N`.
+            If it is ``'mean'``, the output variable holds a scalar value.
+
+    .. admonition:: Example
+
+        In this case, the batch size ``N`` is 4 and the number of classes ``K``
+        is 2.
+        >>> import numpy as np
+        >>> x = np.stack((np.arange(10),5*np.ones((10,))),1).astype(np.float32)
+        >>> t = np.ones((10,),np.int32)
+        >>> hinge_max_margin(x, t, norm='L1', reduce='along_second_axis')
+        variable([0., 0., 0., 0., 0., 1., 2., 3., 4., 5.])
+        >>> hinge_max_margin(x, t)
+        variable(5.5)
+        >>> hinge_max_margin(x, t, norm='L1')
+        variable(1.5)
+        >>> hinge_max_margin(x, t, norm='Huber')
+        variable(1.025)
+
+
+    """
+    return HingeMaxMargin(norm, reduce)(x, t)
+
+
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+u=np.linspace(-4,2)
+m = np.clip((1-u),0,np.inf)
+h = (u>=-1)*m**2/4 + (u<=-1)*(m-1)
+h = (m<2)*m**2/4 + (m>=2)*(m-1)
+
+plt.figure()
+plt.plot(u,h,'r*')
+plt.plot(u,m**2/4)
+plt.plot(u,m)
+
+
+plt.figure()
+plt.plot(u,(m>0)*np.maximum(-m/2,-1),'r*')
+plt.plot(u,(m>0)*-m/2)
+plt.plot(u,-(m>0))
+"""
+
+if __name__ == '__main__':
+    help(hinge_max_margin)
```

Comment:
Could you remove the lines from L:160 to L:181?

Reply:
Done